src/__tests__/patients/care-goals/AddCareGoalModal.test.tsx

```diff
@@ -18,83 +16,65 @@ describe('Add Care Goal Modal', () => {
     jest.resetAllMocks()
   })
 
   it('should render a modal', () => {
-    const { wrapper } = setup()
+    setup()
 
-    const modal = wrapper.find(Modal)
-    const sucessButton = modal.prop('successButton')
-    const closeButton = modal.prop('closeButton')
+    expect(screen.getByRole('dialog')).toBeInTheDocument()
+    expect(screen.getAllByText('patient.careGoal.new')[0]).toBeInTheDocument()
 
-    expect(modal).toHaveLength(1)
-    expect(modal.prop('title')).toEqual('patient.careGoal.new')
-    expect(sucessButton?.children).toEqual('patient.careGoal.new')
-    expect(sucessButton?.icon).toEqual('add')
-    expect(closeButton?.children).toEqual('actions.cancel')
+    expect(screen.getByRole('button', { name: /patient.careGoal.new/i })).toBeInTheDocument()
+    expect(screen.getByRole('button', { name: /actions.cancel/i })).toBeInTheDocument()
   })
 
   it('should render a care goal form', () => {
-    const { wrapper } = setup()
+    setup()
 
-    const careGoalForm = wrapper.find(CareGoalForm)
-    expect(careGoalForm).toHaveLength(1)
+    expect(screen.getByLabelText('care-goal-form')).toBeInTheDocument()
   })
 
   it('should save care goal when save button is clicked and close', async () => {
     const expectedCreatedDate = new Date()
     Date.now = jest.fn().mockReturnValue(expectedCreatedDate)
 
     const expectedCareGoal = {
-      id: '123',
       description: 'some description',
-      startDate: new Date().toISOString(),
-      dueDate: new Date().toISOString(),
-      note: '',
-      priority: 'medium',
-      status: CareGoalStatus.Accepted,
-      achievementStatus: CareGoalAchievementStatus.InProgress,
-      createdOn: expectedCreatedDate,
+      createdOn: expectedCreatedDate.toISOString(),
     }
 
-    const { wrapper } = setup()
-    await act(async () => {
-      const careGoalForm = wrapper.find(CareGoalForm)
-      const onChange = careGoalForm.prop('onChange') as any
-      await onChange(expectedCareGoal)
-    })
+    const { onCloseSpy } = setup()
 
-    wrapper.update()
+    userEvent.type(screen.getAllByRole('textbox')[0], expectedCareGoal.description)
+    userEvent.click(screen.getByRole('button', { name: /patient.careGoal.new/i }))
 
-    await act(async () => {
-      const modal = wrapper.find(Modal)
-      const sucessButton = modal.prop('successButton')
-      const onClick = sucessButton?.onClick as any
-      await onClick()
+    await waitFor(() => {
+      expect(PatientRepository.saveOrUpdate).toHaveBeenCalledTimes(1)
     })
 
-    expect(PatientRepository.saveOrUpdate).toHaveBeenCalledTimes(1)
-    expect(PatientRepository.saveOrUpdate).toHaveBeenCalledWith({
-      ...patient,
-      careGoals: [expectedCareGoal],
-    })
+    expect(PatientRepository.saveOrUpdate).toHaveBeenCalledWith(
+      expect.objectContaining({
+        careGoals: expect.arrayContaining([expect.objectContaining(expectedCareGoal)]),
+      }),
+    )
 
     expect(onCloseSpy).toHaveBeenCalledTimes(1)
```

Comment:
a rerun of all tests passed so not sure. Perhaps it was just a temporary failure.

Reply:
Spy's that check code implementation like that should be replaced with better RTL refactors, nice catch!! 

@nobrayner

Reply:
This one should be fixed now. The test got pulled out of the Modal suite, and moved to the CareGoalTab suite where it can be more easily checked as a user would check.